anaconda_packaging_utils/api/jira_api.py

```diff
@@ -0,0 +1,80 @@
+"""
+File:           jira_api.py
+Description:    Wrapper library that provides tools for using the Python JIRA API. The primary purpose of this wrapper
+                is to simplify and standardize the authentication process.
+
+                JIRA API links:
+                  - Source: https://github.com/pycontribs/jira
+                  - Docs: https://jira.readthedocs.io/index.html#
+"""
+import logging
+from typing import Callable, Final
+
+from jira.client import JIRA
+
+from anaconda_packaging_utils.storage.config_data import ConfigData
+
+# Logging object for this module
+log = logging.getLogger(__name__)
+
+# Where our JIRA boards are hosted
+_JIRA_HOST_URL: Final[str] = "https://anaconda.atlassian.net/"
+
+
+class ApiException(Exception):
+    """
+    Generic exception indicating an unrecoverable failure of this API.
+
+    This exception is meant to condense many possible failures into one generic error. The thinking is, if the calling
+    code runs into any API failure, there isn't much that can be done. So it is easier for the caller to handle one
+    exception than many exception types.
+    """
+
+    def __init__(self, message: str):
+        """
+        Constructs an API exception
+        :param message: String description of the issue encountered.
+        """
+        super().__init__(message if len(message) else "An unknown API issue was encountered.")
+
+
+class JiraApi:
+    """
+    Singleton wrapper to the Python Jira project. This "ensures" that we only construct and authenticate the underlying
+    Jira object once.
+    """
+
+    # The Jira API is wrapped in a list as a cheesy way to work around an initialization problem. Defaulting to `None`
+    # or using an `Optional` causes the static analyzer to freak out on every use of the `_jira`, even if the static
+    # variable has to have been initialized by instance-method call time.
+    _jira: list[JIRA] = []
```

Comment:
```suggestion
    __jira: list[JIRA] = []
```
Double underscore will mangle the make and will make it hard to access the attribute. Someone could have to `var._JiraApi__jira` to access it. See https://docs.python.org/3/tutorial/classes.html#private-variables.

Reply:
TIL, cool!

Reply:
As a heads up, the Google's Pylint file doesn't seem to support this:
```
anaconda_packaging_utils/api/jira_api.py:50:4: C0103: Class attribute name "__jira" doesn't conform to '^(_?[A-Z][A-Z0-9_]*|__[a-z0-9_]+__|_?[a-z][a-z0-9_]*)$' pattern (invalid-name)
```
I'll adjust the file, but I would have expected them to cover this case.

Reply:
Even more interesting, the regex looks like it is meant to handle `__` in a few spots

Reply:
Nvm, I dumped this big ugly regex into an online tool. It'll match `_foo`, `_foo_` and `__foo__` but not `__foo`. Again, weird Google's pylint config wouldn't account for this.